crates/ruma-state-res/src/lib.rs

```diff
@@ -318,7 +318,7 @@ where
             // Only push on the heap once older events have been cleared
             out.remove(node.borrow());
             if out.is_empty() {
-                let (power_level, age) = key_fn(node.borrow())?;
+                let (power_level, age) = key_fn(parent.borrow())?;
```

Comment:
Yikes, this makes more sense now.

Reply:
Yeah. It doesn't help that the naming of this area in the code is all over the place. (In your defense, it's the same in Synapse.) For example, `parent` is actually an event whose `auth_events` contain `node`, rather than the other way around, so I think `child` would be more accurate. I think `zero_outdegree` is also a misnomer (should be more about in-degrees) since the first thing that gets put in there is the `m.room.create` event which will obviously have tons of references.